File: Project Euler/p4.cpp
// Largest palindrome product
#include <bits/stdc++.h>

using namespace std;

bool checkPalindrome(string n) {
    if (n.size() < 2) {
        return true;
    } else {
        char lastDigit = n[n.size()-1];
        char firstDigit = n[0];
        if (firstDigit == lastDigit) {
            return checkPalindrome(n.substr(1, n.size()-2));
        } else {
            return false;
        }
    }
}

int main() {
    vector<int> palindromeInt;
    for (int i = 100; i < 1000; i++) {
        for (int j = 100; j < 1000; j++) {
            if (checkPalindrome(to_string(i*j))) {
                palindromeInt.push_back(i*j);
            }
        }
    }
    cout << *max_element(palindromeInt.begin(), palindromeInt.end()) << endl;
    return 0;
}

File: Project Euler/p4.py
# Largest palindrome product
def check_palindrome(n):
    if len(n) < 2:
        return True
    else:
        last_digit = n[-1]
        first_digit = n[0]
        if first_digit == last_digit:
            return check_palindrome(n[1:-1])
        else:
            return False

def main():
    palindrome_int = []
    for i in range(100, 1000):
        for j in range(100, 1000):
            if check_palindrome(str(i*j)):
                palindrome_int.append(i*j)
    return max(palindrome_int)

if __name__ == "__main__":
    print(main())
